python/ray/tune/suggest/repeater.py

```diff
@@ -0,0 +1,151 @@
+import copy
+import itertools
+import logging
+import numpy as np
+
+from ray.tune.suggest.suggestion import SuggestionAlgorithm
+from ray.tune.experiment import convert_to_experiment_list
+
+logger = logging.getLogger(__name__)
+
+TRIAL_INDEX = "__trial_index__"
+"""str: A constant value representing the repeat index of the trial."""
+
+
+class _TrialGroup:
+    """Internal class for grouping trials of same parameters.
+
+    This is used when repeating trials for reducing training variance.
+
+    Args:
+        primary_trial_id (str): Trial ID of the "primary trial".
+            This trial is the one that the Searcher is aware of.
+        config (dict): Suggested configuration shared across all trials
+            in the trial group.
+        max_trials (int): Max number of trials to execute within this group.
+
+    """
+
+    def __init__(self, primary_trial_id, config, max_trials=1):
+        assert type(config) is dict, (
+            "config is not a dict, got {}".format(config))
+        self.primary_trial_id = primary_trial_id
+        self.config = config
+        self._trials = {primary_trial_id: None}
+        self.max_trials = max_trials
+
+    def add(self, trial_id):
+        assert len(self._trials) < self.max_trials
+        self._trials[trial_id] = None
+
+    def full(self):
+        return len(self._trials) == self.max_trials
+
+    def report(self, trial_id, score):
+        assert trial_id in self._trials
+        assert score is not None, "Internal Error: Score cannot be None."
+        self._trials[trial_id] = score
+
+    def finished_reporting(self):
+        return None not in self._trials.values()
+
+    def scores(self):
+        return list(self._trials.values())
+
+    def count(self):
+        return len(self._trials)
+
+
+class Repeater(SuggestionAlgorithm):
+    """A wrapper algorithm for repeating trials of same parameters.
+
+    It is recommended that you do not run an early-stopping TrialScheduler
+    simultaneously.
```

Comment:
This includes user-specified early-stopping right? (ie through `tune.run(stop=...)`)? can you add a TODO 

shouldn't be too hard to support right, just need to hook into this class when stopping trials?

Reply:
This isn't a missing feature but rather a recommendation. The reason is because if you stop early via scheduling/algorithm, it'll be hard to aggregate the scores.

Reply:
might make sense to support early stopping of entire trial groups though right?